### PR TITLE
Print TLS version & cipher suite when connecting

### PR DESCRIPTION
**Pull request by mcpherrinm**
Todo, tests.

### CODE REVIEW

File: lib/tls.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+var tlsVersions = map[uint16]description{
+	tls.VersionSSL30: {"SSL 3.0", "ssl_3_0", 0},
+	tls.VersionTLS10: {"TLS 1.0", "tls_1_0", 0},
+	tls.VersionTLS11: {"TLS 1.1", "tls_1_1", 1},
+	tls.VersionTLS12: {"TLS 1.2", "tls_1_2", 2},
+}
+
+var cipherSuites = map[uint16]description{
```
Comment (csstaub) [inline lib/tls.go]: ChaCha missing? I would add ChaCha cipher suites here and start requiring Go 1.8 for build.

Comment (mcpherrinm) [inline lib/tls.go]: Sure.  Bumped requirement to 1.8 and added all the new suites.

**Comment by csstaub** [comment]
LGTM. Single nit: README still mentions 1.5 so we should update that

**Comment by mcpherrinm** [comment]
There's a few things I want to do before merging this:

- [ ] Set the "Quality" ratings in the list, so we get red/yellow/green output (all red right now)
- [ ] Come up with more human readable strings
- [ ] Some kind of fallback if it's an unknown cipher suite or tls version

**Comment by mcpherrinm** [comment]
Actually, I'll leave human-readifying the strings til another day.

**Pull request merged**